src/test/java/vvs_dbsetup/CustomersDBTest.java

```diff
@@ -0,0 +1,126 @@
+package vvs_dbsetup;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.*;
+import org.junit.*;
+
+import com.ninja_squad.dbsetup.DbSetup;
+import com.ninja_squad.dbsetup.DbSetupTracker;
+import com.ninja_squad.dbsetup.Operations;
+import com.ninja_squad.dbsetup.destination.Destination;
+import com.ninja_squad.dbsetup.destination.DriverManagerDestination;
+import com.ninja_squad.dbsetup.operation.Operation;
+
+import static vvs_dbsetup.DBSetupUtils.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import webapp.persistence.PersistenceException;
+import webapp.services.*;
+
+/**
+ * Tests the expected behavior of Customers interactions with the database
+ * @author fc45701
+ * @author fc52214
+ */		
+public class CustomersDBTest {
+
+	private static Destination dataSource;
+	
+    // the tracker is static because JUnit uses a separate Test instance for every test method.
+    private static DbSetupTracker dbSetupTracker = new DbSetupTracker();
+	
+    @BeforeClass
+    public static void setupClass() {
+    	startApplicationDatabaseForTesting();
+		dataSource = DriverManagerDestination.with(DB_URL, DB_USERNAME, DB_PASSWORD);
+    }
+    
+	@Before
+	public void setup() throws SQLException, ApplicationException {
+
+		Operation initDBOperations = Operations.sequenceOf(
+			  DELETE_ALL
+			, INSERT_CUSTOMER_SALE_DATA
+			);
+		DbSetup dbSetup = new DbSetup(dataSource, initDBOperations);
+		
+        // Use the tracker to launch the DbSetup. This will speed-up tests 
+		// that do not not change the BD. Otherwise, just use dbSetup.launch();
+        dbSetupTracker.launchIfNecessary(dbSetup);
+		
+	}
+	
+	/**
+	 * a) The SUT does not allow to add a new client with an existing VAT
+	 */
+	@Test
+	public void noDuplicateVATTest() throws ApplicationException {
+		int vat = getFirstCustomerVat();
```

Comment:
If you do it in every method (or more than one) it's useful to move it to the setup - it's always useful to have a valid CUSTOMER_VAT at hand when you're in the test.